### PR TITLE
2FA Refactor

### PR DESCRIPTION
**Pull request by mattlockyer**
Eager to make this repo more modular.
This PR refactors a number of methods previously used by routes defined in app.js into their own modules.
Changes to project structure:
```
/providers
  sms.js
  email.js
/routes
  account.js
  recovery.js
  2fa.js (wip)
```

### CODE REVIEW

**Comment by mattlockyer** [comment]
Tested locally, but not certain if this coverage would be affected by the refactor.

Flags: worried about missing references e.g. a method in runtime not having access to the same variable scope they had when in `app.js`

image

File: app.js [hunk 1]
```diff
@@ -54,221 +54,81 @@ app.use(async (ctx, next) => {
...
-            where: {
-                kind: ctx.request.body.recoveryMethod,
-                publicKey: ctx.publicKey,
-            }
-        });
+        const { kind, publicKey } = ctx.request.body;
+        const [recoveryMethod] = await ctx.account.getRecoveryMethods({ where: {
+            kind: kind,
+            publicKey: publicKey,
+        }});
```
Comment (chadoh) [inline app.js:102]: I would prefer that style changes like this come from a linter and get added all at once; this PR is already going to be pretty big without a bunch of style-change noise

**Comment by vgrichina** [comment]
@mattlockyer overall comments:

- feel like `providers` should just be `utils`, they aren't really providing any objects or smth like that
- what is put in `routes` mostly looks like middleware (`withAccount`, etc) vs specific route handlers
- needs rebase on `master`
- best way to test is actually use it with near-wallet. Unit tests are nice for smoke check and regression testing, but ultimate test is whether this works well with wallet.

**Comment by vgrichina** [comment]
@mattlockyer should we close this one?

**Pull request closed without merging**